rfd/0083-license-expiration-warnings.md

```diff
@@ -0,0 +1,144 @@
+---
+authors: edwarddowling (edward.dolwing@goteleport.com), r0mant (roman@goteleport.com)
+---
+
+
+# RFD 83 - License warnings
+
+## Required approvers
+
+- Engineering: `@r0mant`
+- Product: `@klizentas && @xinding33`
+- Security: `@reedloden`
+
+## What
+
+This RFD proposes a way to display license warnings to self hosted users both before and after license expiration.
+
+## Why
+
+To encourage the renewal of licenses after expiration and let users know before their license expires.
+
+## Success criteria
+
+Teleport displays license warnings to users in both CLI and web UI starting from a certain time prior to the license expiration, as well as post-expiration.
+Teleport administrators can optionally configure the license expiration warning interval but not disable it completely on a per role basis.
+
+## Scope
+
+Self-hosted Teleport will be the target of this RFD. Teleport Cloud is out of scope because the Cloud license does not have proper expiration yet. When Cloud implements proper license expiration, this feature will kick in automatically.
+License warning should be displayed in both Web UI and CLI (tsh/tctl).
+
+### CLI UX
+
+License warnings should be displayed in CLI (tsh/tctl) 
+TSH - on “tsh login”, in “tsh status”
+When the user uses either the “tsh login” or ‘tsh status” commands the appropriate license warning will be displayed. (see examples below)
+TCTL - “tctl status”
+When the user uses the “tctl status” command the appropriate license warning will be displayed.
+After expiration all tsh and tctl will display the license expiry warning.
+
+### Example outputs
+
+The license warning portion of the outputs should be coloured red and orange respectively for the expired and unexpired warnings.
+
+```bash
+$ build/tsh status
+> Profile URL:        https://localhost:9876
+  Logged in as:       someUser
+  Cluster:            someCluster
+  Roles:              access, editor
+  Logins:             someUser
+  Valid until:        2022-08-12 21:57:19 +0100 IST [valid for 4h44m0s]
+  Extensions:         permit-agent-forwarding, permit-port-forwarding, permit-pty
+  Your Teleport Enterprise Edition license will expire in 10 days. Please reach out
+  to   your Teleport Account Manager (or licenses@goteleport.com) to obtain a new
+  license.
```

Comment:
@xinding33 Can you provide a copy for the warning message we want to use here?

Reply:
"Your Teleport Enterprise Edition license will expire in N days. Please reach out to licenses@goteleport.com to obtain a new license. Inaction may lead to unplanned outage or degraded performance and support."